qiime/sdk/tests/test_visualization.py

```diff
@@ -240,6 +241,70 @@ def test_load_from_externally_created_zipfile(self):
             }
             self.assertEqual(fps, expected)
 
+    def test_extract(self):
+        fp = os.path.join(self.test_dir.name, 'visualization.qzv')
+        visualization = Visualization._from_data_dir(self.data_dir,
+                                                     self.provenance)
+        visualization.save(fp)
+
+        output_dir = os.path.join(self.test_dir.name, 'viz-extract-test')
+        result_dir = Visualization.extract(fp, output_dir=output_dir)
+        self.assertEqual(result_dir, output_dir)
+
+        contents = [
+            'visualization/VERSION',
+            'visualization/metadata.yaml',
+            'visualization/README.md',
+            'visualization/data/index.html',
+            'visualization/data/css/style.css']
+        for fp in contents:
+            expected_fp = os.path.join(output_dir, fp)
+            self.assertTrue(os.path.exists(expected_fp),
+                            'File %s was not extracted.' % fp)
+
+    def test_get_index_paths_single_load(self):
+        fp = os.path.join(self.test_dir.name, 'visualization.qzv')
+        visualization = Visualization._from_data_dir(self.data_dir,
+                                                     self.provenance)
+        visualization.save(fp)
+        visualization = Visualization.load(fp)
+
+        actual = visualization.get_index_paths()
+        expected = {'html': 'data/index.html'}
+        self.assertEqual(actual, expected)
+
+    def test_get_index_paths_single_from_data_dir(self):
+        visualization = Visualization._from_data_dir(self.data_dir,
+                                                     self.provenance)
+
+        actual = visualization.get_index_paths()
+        expected = {'html': 'data/index.html'}
+        self.assertEqual(actual, expected)
+
+    def test_get_index_paths_multiple_load(self):
+        most_common_viz(self.data_dir,
```

Comment:
`self.data_dir` contains the output of `mapping_viz` (see `setUp`), which has two files `index.html` and `css/style.css`. Can you write `most_common_viz` output to a different directory under `self.test_dir` in case `mapping_viz` creates other index files in the future?